KSystemInformer/util.c

```diff
@@ -2301,3 +2301,36 @@ NTSTATUS KphDominationAndPrivilegeCheck(
                               ProcessTarget,
                               AccessMode);
 }
+
+/**
+ * \brief Returns the service tag of the current thread on success and 0 on failure.
+ *
+ * \return Service tag .
+ */
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Must_inspect_result_
+ULONG GetCurrentThreadServiceTag(
+    VOID
+    )
+{
+    ULONG SvcTag = 0;
+
+    __try {
+        ULONG_PTR Teb;
+#ifdef _M_ARM64
+        Teb = (*((ULONG_PTR*)(__getReg(18) + 0x30)));
+#elif _WIN64
+        Teb = (ULONG_PTR)__readgsqword(0x30);
+#else ! _WIN64
+        Teb = (ULONG_PTR)__readfsdword(0x18);
+#endif _WIN64
+
+        ULONG* pSvcTag = (ULONG*)((PVOID)(Teb + FIELD_OFFSET(TEB, SubProcessTag)));
```

Comment:
`SubProcessTag` is 64 or 32 bits, I think we should capture the information as it's presented for the sake of future-proofing including `SubProcessTag` in messages. 

Might also be worth calling the API (and message members) it what it is "SubProcessTag" rather than "ServiceTag". I recolonize that's what it is used for, but it might be worth just naming it after what is in the `TEB` for the sake of clarity?

Reply:
yes makes sense, lets go with a 64bit SubProcessTag